RealmSwift/Tests/CombineTests.swift

```diff
@@ -357,6 +357,52 @@ class CombineObjectPublisherTests: CombinePublisherTestCase {
         XCTAssertEqual(prev!.intCol, 100)
     }
 
+    func testChangeSetSubscribeOnKeyPath() {
+        let obj = try! realm.write { realm.create(SwiftObject.self, value: ["intCol": 0, "boolCol": false]) }
+        let sema = DispatchSemaphore(value: 0)
+
+        var prev: SwiftObject?
+        cancellable = changesetPublisher(obj, keyPaths: ["intCol"])
+            .subscribe(on: subscribeOnQueue)
+            .assertNoFailure()
+            .sink(receiveCompletion: { _ in sema.signal() }, receiveValue: { change in
+                if case .change(let o, let properties) = change {
+                    XCTAssertNotEqual(self.obj, o)
+                    XCTAssertEqual(properties.count, 1)
+                    XCTAssertEqual(properties[0].name, "intCol")
+                    if let prev = prev {
+                        XCTAssertEqual(properties[0].oldValue as? Int, prev.intCol)
+                    }
+                    XCTAssertEqual(properties[0].newValue as? Int, o.intCol)
+                    prev = o.freeze()
+                    XCTAssertEqual(prev!.intCol, o.intCol)
+
+                    if o.intCol == 100 {
+                        sema.signal()
+                    }
+                } else {
+                    XCTFail("Expected .change but got \(change)")
+                }
+            })
+
+        // The following line checks to make sure a change not
+        // within the intended keyPath does *not* publish a
+        // change.
+        // If a changeset is published for boolCol, the test would fail
+        // above when checking for property name.
+        try! realm.write { obj.boolCol = true }
```

Comment:
I think making this write which should not produce a notification in between writes which should would do a better job of catching potential bugs than having it entirely before them.

Reply:
The github ui doesn't show this comment as out-dated, but this change should've been made in the latest commit. 
`try! realm.write { obj.boolCol = true }` is no longer the leading write.

Reply:
I just want to confirm that the current ordering doesn't satisfy your last comment (because I think the github review ui is slightly off). Here is the [file](https://github.com/realm/realm-cocoa/blob/em/kpf/RealmSwift/Tests/CombineTests.swift).
As of `b9bc111`, the ordering of the writes are:
1) [expect a change](https://github.com/realm/realm-cocoa/blob/em/kpf/RealmSwift/Tests/CombineTests.swift#L389)
2) [expect no change](https://github.com/realm/realm-cocoa/blob/em/kpf/RealmSwift/Tests/CombineTests.swift#L398)
3) [expect a change](https://github.com/realm/realm-cocoa/blob/em/kpf/RealmSwift/Tests/CombineTests.swift#L400)

So the write which expects no change would be in front and behind of writes which expect changes.
If this is _not_ what you're talking about, then I can create two tests where the writes are ordered as:
1) expect a change
2) expect no change

1) expect no change
2) expect a change

Reply:
To detect that a notification was not sent, we need to receive a notification which would have included the change we're filtering out had it not been filtered out. Deleting the object doesn't do this: with no keypath filtering involved, mutating a property and then deleting the object will result in a notification which just says that the object was deleted unless you do something which ensures that the two operations cannot be batched together. This requires not only that they be in separate write transactions but that there is a synchronization point where you can guarantee that all previous writes have been fully processed before you delete the object.

The simplest way to do this is to mutate intCol again after the mutation on boolCol so that you can wait for the notification for that write and verify that it does not mention boolCol.